camel-chimera-graph/src/test/java/com/cefriel/GraphSparqlTest.java

```diff
@@ -0,0 +1,176 @@
+        for(BindingSet binding: result) {
+            assert (binding.getBindingNames().size() == 3);
+            assert (binding.getBindingNames().containsAll(List.of("s", "p", "o")));
+        }
+    }
+
+    @Test
+    public void testSparqlAskQuery() throws Exception {
+        var graph = new MemoryRDFGraph();
+        Utils.populateRepository(graph.getRepository(), triples, null);
+
+        MockEndpoint mock = getMockEndpoint("mock:sparqlAskQuery");
+        mock.expectedMessageCount(0);
+        mock.assertIsSatisfied();
+
+        Assertions.assertThrows(CamelExecutionException.class,
+                () -> {
+                    try {
+                        start.sendBody("direct:sparqlAskQuery", graph);
+                    }
+                    catch (CamelExecutionException e) {
+                        Throwable cause = e.getCause();
+                        Assertions.assertThrows(IllegalArgumentException.class,
+                                () -> {throw cause;});
+                        throw e;
+                    }
+                });
+    }
+
+    @Test
+    public void testSparqlResourceQuery() throws Exception {
+        var graph = new MemoryRDFGraph();
+        Utils.populateRepository(graph.getRepository(), triples, null);
+        start.sendBody("direct:sparqlResourceQuery", graph);
+
+        MockEndpoint mock = getMockEndpoint("mock:sparqlResourceQuery");
+        mock.expectedMessageCount(1);
+        mock.assertIsSatisfied();
+
+        List<BindingSet> result = (List<BindingSet>) mock.getExchanges().get(0).getMessage().getBody();
+        for(BindingSet binding: result) {
+            assert (binding.getBindingNames().size() == 3);
+            assert (binding.getBindingNames().containsAll(List.of("s", "p", "o")));
+        }
+    }
+
+    @Test
+    public void testSparqlQueryUnsupportedFormat() throws Exception {
+        var graph = new MemoryRDFGraph();
+        Utils.populateRepository(graph.getRepository(), triples, null);
+
+        Assertions.assertThrows(CamelExecutionException.class,
+                () -> {
+                    try {
+                        start.sendBody("direct:sparqlQueryUnsupportedFormat", graph);
+                    }
+                    catch (CamelExecutionException e) {
+                        Throwable cause = e.getCause();
+                        Assertions.assertThrows(IllegalArgumentException.class,
+                                () -> {throw cause;});
+                        throw e;
+                    }
+                });
+
+        MockEndpoint mock = getMockEndpoint("mock:sparqlQueryUnsupportedFormat");
+        mock.expectedMessageCount(0);
+        mock.assertIsSatisfied();
+    }
+
+    @Test
+    public void testSparqlQueryNoQueries() throws Exception {
+        var graph = new MemoryRDFGraph();
+        Utils.populateRepository(graph.getRepository(), triples, null);
```

Comment:
These `utils` methods used for testing should be moved to a separate class under the `test` package

Reply:
Not sure of what you mean, what `util` methods?